src/main/java/javaslang/collection/IndexedSeq.java

```diff
@@ -346,3 +351,66 @@ default boolean startsWith(Iterable<? extends T> that, int offset) {
     IndexedSeq<Tuple2<T, Integer>> zipWithIndex();
 
 }
+
+interface IndexedSeqModule {
+    class LastIndexOfSlice {
+        static <T> int lastIndexOfSlice(IndexedSeq<T> t, IndexedSeq<T> slice, int end) {
+            if (end < 0) {
+                return -1;
+            }
+            if (t.isEmpty()) {
+                return slice.isEmpty() ? 0 : -1;
+            }
+            if (slice.isEmpty()) {
+                int len = t.length();
+                return len < end ? len : end;
```

Comment:
Ah - I think I've not understood/overseen `end` - we search until there. It is ok!